### ISSUE TITLE
Rename `components` directory to `resources` under .dapr directory

### ISSUE THREAD

**Issue by pravinpushkar** [dapr/cli#1148]
## Describe the proposal

The issue #953 layout a proposal for using `resources-path` flag with `dapr run` instead of `components-path` flag and slowly deprecating it.
In the similar context it would make sense to rename `components` directory to `resources` and slowly deprecate `components` directory use. 

Since the `components` directory is heavily used and exists from beginning so, migrating from `components` to `resources` directly should be done in a phased manner.

We can use this issue to discuss different approaches for this issue.

## Release Note

RELEASE NOTE:

**Comment by pravinpushkar** [comment]
@mukundansundar @yaron2 @shubham1172 @msfussell

**Comment by shubham1172** [comment]
Sounds good!
1. When running a `dapr init`, use `resources` instead of `components`.
1. When running `dapr run`, order of precedence can be `resources` > `components`. If `components` is used, a warning log can be added with a link to the GitHub issue with more details on the deprecation.

**Comment by mukundansundar** [comment]
> Sounds good!
> 
> 1. When running a `dapr init`, use `resources` instead of `components`.
> 2. When running `dapr run`, order of precedence can be `resources` > `components`. If `components` is used, a warning log can be added with a link to the GitHub issue with more details on the deprecation.

Agreed. Only when default dir is used, will the deprecation notice be printed i.e. for `~/.dapr/components` dir.
Also when `--components-path` is used, then also a deprecation notice needs to be printed.

**Comment by shubham1172** [comment]
> There might be a preexisting ~/.dapr/components if dapr uninstall --all is not used before dapr init

In this case, shall we warn and copy files to `resources`?

**Comment by pravinpushkar** [comment]
> Out of scope for this issue?

It is being covered in #1012

**Comment by mukundansundar** [comment]
> Based on the above suggestions -
> 
> 1. When User just upgrades his `dapr cli` without uninstalling -> all usage of `dapr run --components-path ""` will start giving deprecation message but continues to work.
> 2. User does `dapr init` after `dapr uninstall` or `dapr uninstall  --all` -> Create `resources` directory always and copy the files from `components` directory if `components` dir exists.
> 3. In all cases `dapr run` without either of the above flags, will default to looking `resources` dir first then `components` dir.

this looks good to me.
@yaron2 thoughts?

**Comment by mukundansundar** [comment]
Have triaged this to be completed in 1.11
